### PR TITLE
fixing mid_point on Julia 1.0

### PR DESCRIPTION
**Pull request by mateuszbaran**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -4,7 +4,9 @@ using ManifoldsBase
 
     num_ambiguities = length(Test.detect_ambiguities(ManifoldsBase))
```
Comment (kellertuer) [inline test/runtests.jl:5]: In Manifolds even the call here failed, so would it maybe be better to even put this line into the version test?

File: src/ManifoldsBase.jl [hunk 2]
```diff
@@ -565,8 +565,19 @@ Saves the result in `q`.
 """
 function mid_point!(M::Manifold, q, p1, p2)
     X = log(M, p1, p2)
-    X /= 2
-    return exp!(M, q, p1, X)
+    return exp!(M, q, p1, X / 2)
+end
+
+@static if VERSION <= v"1.1"
```
Comment (kellertuer) [inline src/ManifoldsBase.jl:571]: Why does this not work? It looks like exp/log should work anyways and despite that we only divide by two? Maybe `fill(X / 2)` could just be `X ./ 2`?

File: test/runtests.jl (hunk 1 above)
Comment (mateuszbaran) [reply test/runtests.jl:5]: Are you sure? Just calculating the number of ambiguities shouldn't fail.

Comment (kellertuer) [reply test/runtests.jl:5]: Well, it was with `Test.detect_ambiguities(Base)` which ran into a Stack Overflow, so in `Manifolds` also that call is within the `if`.

Comment (kellertuer) [reply test/runtests.jl:5]: But it's also not that bad, so do what you feel is better; I'll already accept this :)

Comment (mateuszbaran) [reply test/runtests.jl]: OK, I've put it in that `if`.

**Comment by kellertuer** [comment]
Oh and could you write a short test for the new case?

**Comment by mateuszbaran** [comment]
I've added a test but these lines won't be detected as covered by codecov since coverage data is generated on a different version of Julia.

**Comment by kellertuer** [comment]
You're right, well, still good to have the test.

**Pull request merged**